documentation/docs/tutorials/setup.md

```diff
@@ -528,3 +528,142 @@ The Grafana dashboard should be accessible at `http://<your-ip>:3000`.
 Default login credentials are:
 * `username`: admin
 * `password`: admin
+## Installing Goshimmer by Building From Source
+### Software Requirements
+Upgrade your systems' packages by running the following command:
+```bash
+apt update && apt dist-upgrade -y
+```  
+
+#### Installing RocksDB Compression Libraries
+
+GoShimmer uses RocksDB as its underlying database engine. You can install the dependencies required by RocksDB by running the following command:
+
+```bash
+sudo apt-get install libgflags-dev libsnappy-dev zlib1g-dev libbz2-dev liblz4-dev libzstd-dev
+```  
+
+
+####  GCC and G++
+
+GCC and G++ are required for the compilation to work properly.  You can install them by running the following command:
+
+```bash
+sudo apt install gcc g++
+```
+
+### Clone the Repository
+
+Once you have installed the [software requirements](#software-requirements), you should clone the [GoShimmer repository](https://github.com/iotaledger/goshimmer/) into the `/opt` directory. You can do so by running the following commands: 
+
+```bash
+cd /opt
+git clone https://github.com/iotaledger/goshimmer.git
+```
+
+###  Golang-go 1.16
```

Comment:
The whole Golang-go installation section would probably make more sense as a child of ### Software Requirements